rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+        throw new InvalidRCodeException("Not enough bytes to reach type block");
+      }
+      processTypeBlock(input, out);
+      isClass = true;
+    }
+
+    input.close();
+  }
+
+  private final void processHeader(InputStream input, PrintStream out) throws IOException, InvalidRCodeException {
+    byte[] header = new byte[HEADER_SIZE];
+    int bytesRead = input.read(header);
+    if (bytesRead != HEADER_SIZE) {
+      throw new InvalidRCodeException("Not enough bytes in header");
+    }
+
+    if (out != null) {
+      out.printf("%n******%nHEADER%n******%n");
+      printByteBuffer(out, header);
+    }
+
+    long magic = ByteBuffer.wrap(header, HEADER_OFFSET_MAGIC, Integer.BYTES).getInt();
+    if (magic == MAGIC1) {
+      order = ByteOrder.BIG_ENDIAN;
+    } else if (magic == MAGIC2) {
+      order = ByteOrder.LITTLE_ENDIAN;
+    } else {
+      throw new InvalidRCodeException("Can't find magic number");
+    }
+
+    version = ByteBuffer.wrap(header, HEADER_OFFSET_RCODE_VERSION, Short.BYTES).order(order).getShort();
+    sixtyFourBits = (version & 0x4000) != 0;
+    if ((version & 0x3FFF) >= 1100) {
+      timeStamp = ByteBuffer.wrap(header, HEADER_OFFSET_TIMESTAMP, Integer.BYTES).order(order).getInt();
+      md5 = ByteBuffer.wrap(header, HEADER_OFFSET_MD5, Short.BYTES).order(order).getShort();
+      segmentTableSize = ByteBuffer.wrap(header, HEADER_OFFSET_SEGMENT_TABLE_SIZE, Short.BYTES).order(order).getShort();
+      signatureSize = ByteBuffer.wrap(header, HEADER_OFFSET_SIGNATURE_SIZE, Integer.BYTES).order(order).getInt();
+      typeBlockSize = ByteBuffer.wrap(header, HEADER_OFFSET_TYPEBLOCK_SIZE, Integer.BYTES).order(order).getInt();
+      rcodeSize = ByteBuffer.wrap(header, HEADER_OFFSET_RCODE_SIZE, Integer.BYTES).order(order).getInt();
+    } else {
+      throw new InvalidRCodeException("Only v11 rcode is supported");
+    }
+  }
+
+  private final void processSignatureBlock(InputStream input, PrintStream out) throws IOException, InvalidRCodeException {
+    byte[] header = new byte[signatureSize];
+    int bytesRead = input.read(header);
+    if (bytesRead != signatureSize) {
+      throw new InvalidRCodeException("Not enough bytes in signature block");
+    }
+    if (out != null) {
+      out.printf("%n*********%nSIGNATURE%n*********%n");
+      printByteBuffer(out, header);
+    }
+
+    int preambleSize = readAsciiEncodedNumber(header, 0, 4);
+    int numElements = readAsciiEncodedNumber(header, 4, 4);
+    // Version of signature block : offset 8, 4 bytes
+    // Encoding : offset 12, null-terminated string
+
+    int pos = preambleSize;
+    for (int kk = 0; kk < numElements; kk++) {
+      String str = readNullTerminatedString(header, pos);
+      pos += str.length() + 1;
+
+      // Datasets and temp-tables not read for now
+      if (str.startsWith("DSET") || str.startsWith("TTAB")) {
+        continue;
+      }
+
+      // Will probably be skipped
+      // Function fn = parseProcSignature(str);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ACommentedOutCodeLine)